### PR TITLE
Source level

### PR DESCRIPTION
**Pull request by joaobrittoneto**
Need test and integration on syskit.
Set the source level (the signal power). Check depth of vehicle to not set a high power signal in the surface.
Add messages counters.
Use device settings as default property

### CODE REVIEW

File: tasks/UsblAUV.cpp [hunk 1]
```diff
@@ -39,6 +39,8 @@ bool UsblAUV::configureHook()
         }
     }
 
+    temp_source_level = IN_AIR;
```
Comment (doudou) [inline tasks/UsblAUV.cpp]: Really find a better name (e.g. source_level_in_water ?)

**Comment by joaobrittoneto** [comment]
I totally misunderstood the delivery report thing. There is just ONE delivery report per Instant Message (Failed or Delivered). Instant Message's counter make it clear (messageSent = messageDelivered + messageFailed).
Good thing, the code is cleaner. 
The source level on auv and control according its depth needs a better understanding

**Comment by doudou** [comment]
If we have to change the source level dynamically, it would be a lot better to make it a separate dynamic property

**Comment by joaobrittoneto** [comment]
Ok.
@doudou, I'm not sure the best way to deal with the source level, especially in auv, regarding the distance of the auv from the surface and from the dock side.

The Source Level is the power of the signal.  The manual says to set the source level to it minimal level outside water, once it can damage the device. Also, long distances between auv and dock requires a higher source level for data exchange.

There is this other property in usbl, Source Level Control, a bool. If it is set, it will change the local source level to match with the source level of the remote device in case of raw data transmission. 
It's practical so we don't need to check if the actual source level is enough or not. Usbl deals with it.

But regarding the depth, and making sure the usbl will not work with high source level in the surface, how would be a good way to do it?
Change the source level and source level control properties so it stay in its minimal value near surface  and continue to transmit data ( even if it may be not delivered to remote device)?
Or just drop the data and not send it (raw data and instant message) when auv is above a certain depth (like 0.5m), and avoid change the properties with the change of depth?
What do you suggest?

**Comment by doudou** [comment]
If you make the source level a dynamic property, we could change it dynamically in Syskit without a problem. Just make the source level minimal by default.

Now, this source level control thing looks appealing. However, the AUV is really better placed than the dock to decide the source level (since it has an estimate of its own position).... So I would go with manually controlling it inside Syskit.

**Comment by joaobrittoneto** [comment]
Ok. So it's not necessary to monitor the depth and change the parameter direct inside the task? It can be done by syskit through the dynamic property?

About the source level control, I think it should be set in one of the nodes. If not in auv, at least in dock.  
If we control the source level in auv, then we need to monitor not only the depth, but also the quality of the acoustic link and estimate the distance to dock, and I think it can be a though task.  If we do it manually in auv, the vehicle need to be tethered.

We can do some tests till we better understanding the boundaries of each source level and when auv runs untethered, it automatically defines its source level. But if this automatic method fails we risk not been able to connect with auv.  

So, I would go for the dock side defines the source level and let the source level control set in auv.
For safety, both the source level and source level control would be dynamic properties and auv just monitor the depth so it can unset the source level control and set the source level to minimal near surface.  
What do you think?

**Comment by joaobrittoneto** [comment]
Source Level and Source Level Control as dynamic properties.

**Comment by doudou** [comment]
> So, I would go for the dock side defines the source level and let the source level control set in auv.
> For safety, both the source level and source level control would be dynamic properties and auv just monitor the depth so it can unset the source level control and set the source level to minimal near surface.

I think it's probably the best option, at least until we get a better idea of how well / how all that stuff works. What I am concerned about is what happens when we lose the link. When that happens, the AUV will obviously not be able to do level control, so what happens then.

Anyways, let's use the automated way and figure it out when we get some experience.

**Pull request merged**